### PR TITLE
6. PATCH /api/reviews/:review_id

### PR DESCRIPTION
**Pull request by MartinSWDev**
tests and functionality for patch added, changed error handler for code 22P02 to handle all syntax errors

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -158,5 +158,135 @@ describe('app.js tests', () => {
...
+        })
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toEqual('Invalid input syntax');
+        });
+    });
+    test('400: Id is too large', () => {
+      return request(app)
+        .patch('/api/reviews/1010101010101')
+        .send({
```
Comment (Rayhaan2k9) [inline __tests__/app.test.js:272]: I don't think you need this test. This should fall under 404 as you could have a review with id 1010101010101 in the future!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -158,5 +158,135 @@ describe('app.js tests', () => {
...
+        })
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.updated.votes).toBe(0);
+        });
+    });
+    test('object with negative value cant go below 0', () => {
+      return request(app)
+        .patch('/api/reviews/1')
+        .send({
```
Comment (Rayhaan2k9) [inline __tests__/app.test.js:205]: You don't need this logic. There could be reviews with votes in the minuses if they are poor reviews! 😅

File: models/patchReviewById.model.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+const db = require('../db/connection');
+exports.patchReview = (reviewId, voteChange) => {
+  if (reviewId >= 2147483647) {
+    return Promise.reject({
```
Comment (Rayhaan2k9) [inline models/patchReviewById.model.js:4]: We don't need this logic if we are removing the test I commented on

File: __tests__/app.test.js (hunk 1 above)
Comment (MartinSWDev) [reply __tests__/app.test.js:272]: number is too large for INT datatype in SQL so throws a 400 error but I wouldn't put it in the same class as "invalid syntax" as it IS a number still

Comment (Rayhaan2k9) [reply __tests__/app.test.js:272]: Ahh that makes sense! Ignore me then!

**Review by Rayhaan2k9 — APPROVED** [review]
I would maybe change the minus votes logic you have but great work!

**Comment by MartinSWDev** [comment]
Have removed below 0 limit and tests for that functionality

**Pull request merged**